pkg/pillar/cmd/zedrouter/ipam.go

```diff
@@ -68,6 +91,17 @@ func (z *zedrouter) generateAppMac(appUUID uuid.UUID, adapterNum int, appNum int
 	return nil
 }
 
+func (z *zedrouter) generateAppMac(appUUID uuid.UUID, adapterNum int, appNum int,
+	netInstStatus *types.NetworkInstanceStatus) net.HardwareAddr {
+	if z.legacyMACAddr {
```

Comment:
One important thing to note is that we have not changed how MAC addresses are generated for switch network instances, only for local (see the code before and after, lines are rearranged but the logic have not changed for the switch case). Meaning that this `if z.legacyMACAddr` condition could be under `case types.NetworkInstanceTypeLocal` and I would also avoid this log messages about legacy/random MAC address for switch NI.
Moreover, I would suggest to name the config option: `network.local.legacy.mac.address` (similar to `network.switch.enable.arpsnoop` which we have for switch NI). Also, I would not put `zedrouter` there, as we do not have names of microservices in other config properties either.

Reply:
Thanks, will change accordingly.